api/net/Areas/Editor/Controllers/StorageController.cs

```diff
@@ -49,216 +55,627 @@ public StorageController(IOptions<StorageOptions> options)
+                if (safePath.DirectoryExists()) throw new InvalidOperationException("Invalid path");
 
-        if (!overwrite && safePath.FileExists()) throw new InvalidOperationException("File already exists");
+                var directory = Path.GetDirectoryName(safePath);
+                if (directory?.DirectoryExists() == false)
+                    Directory.CreateDirectory(directory);
 
-        // TODO: Handle multiple files.
-        using var stream = System.IO.File.Open(safePath, FileMode.Create);
-        await file.CopyToAsync(stream);
+                if (!overwrite && safePath.FileExists()) throw new InvalidOperationException("File already exists");
 
-        return new JsonResult(new ItemModel(safePath));
+                // TODO: Handle multiple files.
+                using var stream = System.IO.File.Open(safePath, FileMode.Create);
+                await file.CopyToAsync(stream);
+
+                return new JsonResult(new ItemModel(safePath));
+            }
+            else throw new NotImplementedException($"Location connection type '{connection?.ConnectionType}' not implemented yet.");
+        }
+        else
+        {
+            var safePath = Path.Combine(_options.GetUploadPath(), path, file.FileName);
+            if (String.IsNullOrWhiteSpace(Path.GetFileName(safePath))) throw new InvalidOperationException("Filename missing");
+            if (safePath.DirectoryExists()) throw new InvalidOperationException("Invalid path");
+
+            var directory = Path.GetDirectoryName(safePath);
+            if (directory?.DirectoryExists() == false)
+                Directory.CreateDirectory(directory);
+
+            if (!overwrite && safePath.FileExists()) throw new InvalidOperationException("File already exists");
+
+            // TODO: Handle multiple files.
+            using var stream = System.IO.File.Open(safePath, FileMode.Create);
+            await file.CopyToAsync(stream);
+
+            return new JsonResult(new ItemModel(safePath));
+        }
     }
 
     /// <summary>
     /// Stream the file for the specified path.
     /// </summary>
+    /// <param name="locationId"></param>
     /// <param name="path"></param>
-    /// <param name="location"></param>
     /// <returns></returns>
     [AllowAnonymous] // TODO: Temporary to test HTML 5 video
     [HttpGet("stream")]
-    [HttpGet("{location}/stream")]
+    [HttpGet("{locationId:int}/stream")]
     [Produces("application/octet-stream")]
     [ProducesResponseType(typeof(FileStreamResult), (int)HttpStatusCode.OK)]
     [ProducesResponseType(typeof(FileStreamResult), (int)HttpStatusCode.PartialContent)]
     [ProducesResponseType((int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "Storage" })]
-    public IActionResult Stream(string path, [FromRoute] string? location = "capture")
+    public IActionResult Stream([FromRoute] int? locationId, [FromQuery] string path)
```

Comment:
I wish we could stream from the remote location, however it loses the connection.  The only solution so far is to copy it locally first.